### PR TITLE
IGNORE-add func test for rejected env.

### PR DESCRIPTION
**Pull request by mdayican**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/BPS-1039

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/CrimeBlobDispatchingTest.java [hunk 1]
```diff
@@ -59,4 +63,40 @@ void should_move_extracted_crime_envelope_to_crime_storage() throws Exception {
...
         );
     }
+
+    @Test
+    void should_reject_invalid_crime_envelope() throws Exception {
+        // upload crime file with unique name
+        String fileName = randomFileName();
+
+        byte[] wrappingZipContent = createZipArchive(
+            asList("test-data/envelope/envelope.zip")
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/CrimeBlobDispatchingTest.java]: may be use an invalid zip file with name suggesting it has invalid signature ?

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/CrimeBlobDispatchingTest.java [hunk 2]
```diff
@@ -59,4 +63,39 @@ void should_move_extracted_crime_envelope_to_crime_storage() throws Exception {
             internalZipContent
         );
     }
+
+    @Test
+    void should_reject_invalid_crime_envelope() throws Exception {
```
Comment (nitinprabhu) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/CrimeBlobDispatchingTest.java]: should_reject_envelope_with_invalid_signature_and_move_to_rejected_container ?

**Comment by lgonczar** [comment]
If we're creating this new test in crime tests, should we also create a similar one for bulk scan ones? Or maybe we should create that test in its own test class?

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import static org.hamcrest.Matchers.equalTo;
+import static org.springframework.http.HttpStatus.OK;
+import static uk.gov.hmcts.reform.blobrouter.data.model.Status.REJECTED;
+import static uk.gov.hmcts.reform.blobrouter.envelope.ZipFileHelper.createZipArchive;
+import static uk.gov.hmcts.reform.blobrouter.storage.StorageHelper.blobExists;
+import static uk.gov.hmcts.reform.blobrouter.storage.StorageHelper.uploadFile;
+
+public class BlobRejectTest extends FunctionalTestBase {
+
+    private BlobServiceClient crimeStorageClient;
```
Comment (lgonczar) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: We can get rid of this client completely.

**Comment by rkondratowicz** [comment]
> If we're creating this new test in crime tests, should we also create a similar one for bulk scan ones? Or maybe we should create that test in its own test class?

maybe separate class with a test that uploads to several containers?

File: src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+            .queryParam("file_name", fileName)
+            .queryParam("container", config.crimeSourceContainer)
+            .get("/envelopes")
+            .then()
+            .statusCode(OK.value())
+            .body("container", equalTo("crime"))
+            .body("status", equalTo(REJECTED.name()))
+            .body("is_deleted", equalTo(true));
+
+        byte[] expectedContent = toByteArray(getResource("test-data/envelope/envelope.zip"));
```
Comment (rkondratowicz) [inline src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: isn't expected content `wrappingZipContent`? (that's what you upload, not the inner envelope)

Comment (mdayican) [reply src/functionalTest/java/uk/gov/hmcts/reform/blobrouter/BlobRejectTest.java]: please ignore this pr for now.

**Comment by mdayican** [comment]
closing this. https://github.com/hmcts/blob-router-service/pull/229
and 
https://github.com/hmcts/blob-router-service/pull/231 
created

**Pull request closed without merging**